app.rb

```diff
@@ -0,0 +1,167 @@
+require_relative 'book'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def run
+    puts 'Welcome to School Library App!'
+    menu
+  end
+
+  def menu
+    loop do
+      puts ' '
+      puts 'Please choose an option by entering a number:'
+      puts '1 - List all books'
+      puts '2 - List all people'
+      puts '3 - Create a person'
+      puts '4 - Create a book'
+      puts '5 - Create a rental'
+      puts '6 - List all rentals for a given person id'
+      puts '7 - Exit'
+      print ''
+
+      choice = gets.chomp.to_i
+
+      case choice
+      when 1..6 then options(choice)
+      when 7 then puts 'Thank you for using our app.'
+      end
+      break if choice == 7
+    end
+  end
+
+  def options(choice)
+    case choice
+    when 1
+      list_books
+    when 2
+      list_people
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      list_rentals
+    end
+  end
```

Comment:
- [ ] Please move this list of options into your main.rb file as it should deal with the user interactions only, all the methods should belong to the app class as you did already.

Reply:
Hey, options function is called inside the menu function, it wouldn't be wise to move that unless it is a must requirement.